### PR TITLE
[Merged by Bors] - chore: Delete `Init.Data.Subtype.Basic`

### PR DESCRIPTION
**Pull request by YaelDillies**
The few useful lemmas can go to `Data.Subtype.Basic` and the other ones can be deleted.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Subtype.lean [hunk 1]
```diff
@@ -29,6 +30,13 @@ namespace Subtype
 
 variable {α β γ : Sort*} {p q : α → Prop}
 
+#align subtype.eq Subtype.eq
+#align subtype.eta Subtype.eta
+
+#noalign subtype.tag_irrelevant
+#noalign subtype.exists_of_subtype
+#noalign subtype.inhabited
```
Comment (fpvandoorn) [inline Mathlib/Data/Subtype.lean:38]: I believe we don't like `#noalign`s, and prefer lemmas that roughly match, e.g. `Subtype.instInhabitedSubtype` (which probably matches exactly?)

**Comment by fpvandoorn** [comment]
LGTM besides the CI error,

bors d+

File: Mathlib/Data/Subtype.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Data/Subtype.lean:38]:
```suggestion
#align subtype.inhabited Subtype.instInhabitedSubtype
```
(how is that even an instance, though?)

Comment (fpvandoorn) [reply Mathlib/Data/Subtype.lean]: Fair point, I don't think it can ever be succesfully applied.

**Comment by YaelDillies** [comment]
bors merge

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**